version: '3.7'
services:
  nginx:
    container_name: core_web
    restart: on-failure
    image: jonasal/nginx-certbot:latest
    env_file:
      - .env.nginx.local
    volumes:
      - nginx_secrets:/etc/letsencrypt
      - ./nginx/nginx.local.conf:/etc/nginx/conf.d/default.conf
    ports:
      - "80:80"
      - "443:443"
    depends_on:
      - web
  web:
    container_name: core_app
    build:
      context: .
      dockerfile: ./Dockerfile
    restart: always
    env_file: .env.local
    ports:
      - "8000:8000"
    command:
      - sh
      - -c
      - |
        python manage.py migrate &&
        gunicorn modu_property.wsgi --bind 0.0.0.0:8000 --workers 1 --threads 1
    volumes:
     - .:/app
     - static_volume:/app/static
    depends_on:
     - db
  celery-worker:
    container_name: celery-worker
    build:
      context: .
      dockerfile: ./Dockerfile
    restart: always
    volumes:
      - .:/var/www/app/worker
    command:
      - sh
      - -c
      - celery -A modu_property worker -l INFO --concurrency 1 -E
      - tail -f /dev/null
    depends_on:
      - web
      - redis
  celery-beat:
    container_name: celery-beat
    build:
      context: .
      dockerfile: ./Dockerfile
    volumes:
      - .:/var/www/app/beat
    command:
      - sh
      - -c
      - celery -A modu_property beat -l INFO
      - tail -f /dev/null
    links:
      - celery-worker
      - redis
    depends_on:
      - web
      - redis
      - celery-worker
  redis:
    restart: unless-stopped
    image: redis:7.0.5-alpine
    expose:
      - 6379
  db:  # db로 하면 도커가 아닌 파이참에서 서버 띄울 때 .env.local에서 DB_HOST 값이 db라 문제 생김. 임의로 localhost로 이름 붙임.
    container_name: core_db
    image: postgres:14.6-alpine
    env_file: .env.local
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    ports:
      - '5432:5432'
    user: postgres
    environment:
      - "POSTGRES_DB=modu_property"
      - "POSTGRES_USER=postgres"
      - "POSTGRES_PASSWORD=postgres"
 
volumes:
  static_volume:
  postgres_data:
  nginx_secrets:
